Add dtwclust package

bloodearnest opened the pull request:
This required squashing the image to a single layer, as we'd hit the
limit of 128 layers. The squash.sh script included here performs this.

If we add another 126 packages and need to squash again before we
properly fix the R image, I will cry.

evansd commented on packages.csv:
```diff
@@ -14,7 +14,7 @@
 "MASS","7.3-52"
 "MatchIt","4.2.0"
-"Matrix","1.2-18"
+"Matrix","1.5-1"
```
I wonder if we need to check whether this version change has any significance?

remlapmot replied:
To get a feel for this you can view the Matrix package *NEWS* file at

https://cran.r-project.org/web/packages/Matrix/news.html

It's hard to tell if there are any important changes between the versions. My guess is that this probably won't cause any problems.

This has happened because the current version of dtwclust requires Matrix >=1.5-0, which you can see set in the *Imports:* list here https://cran.r-project.org/package=dtwclust 

In fact a minimum Matrix version was only set in this latest version of dtwclust; so you'd only have to go back 1 version of dtwclust in order to stop Matrix being updated. You can see this easier on the GitHub CRAN mirror (which puts in version updates as single commits) https://github.com/cran/dtwclust/commit/5f65ddd00bb4c24923229e85df33d2861abe728b 


(Note that this isn't the first time a package has been updated in the container; *backports* was updated in 80d5b162d4c7ce86bbfc9c6e3e86a4cc186f3f4d which I think appears not to have caused any problems.)

evansd replied:
Thanks @remlapmot, we were just discussing this internally. I found the news file and this section here gave me pause:

> Changes in version 1.3-0 (2020-12-15 r3351)
> Significant User-Visible Change
> ...

I didn't realise the version requirement was such a recent change in `dtwclust` though — thanks very much for pointing that out. If we can get away with a slightly older version of `dtwclust` then I'd be inclined to go with that rather than risk the upgrade. (Unless people more experienced with R and it's use within OpenSAFELY are confident it's harmelss.) I'll check with the people who requested it.

As you're well aware, the whole R image needs significant work anyway so I'd rather keep changes minimal :smile:

remlapmot replied:
There are several ways to install the previous version.

I'd say the easiest way is to note from its CRAN page the date the current version was released (2022-09-24) and use the Microsoft CRAN snapshot mirror for the day prior to that as the repository in `install.packages()` in R, i.e.,

```r
install.packages('dtwclust', repos = 'https://cran.microsoft.com/snapshot/2022-09-23/')
```
(So you'd have to do this interactively in the container rather using Simon's scripts)

evansd replied:
Preempting my next question! Thanks Tom

bloodearnest replied:
> (So you'd have to do this interactively in the container rather using Simon's scripts)

Challenge accepted!

bloodearnest commented:
I have updated to use the previous release of dwtclust, using @remlapmot's trick, and it's not upgraded any packages.

I've updated the add-pkg.sh script to handle this repo worflow in the process.
